Httplib load integration (#4099)
mlpack/mlpack: doc/user/load_save.md

## `Load()`

 - `Load(filename, object)`

>    * Load `object` from the given file `filename` with default options:

Author: In this case, all should be X, t is too confusing to have both of them combined

    << std::endl;
```

Another simple example, loading the file from a URL:

```c++
arma::mat data;
bool success = mlpack::Load("http://datasets.mlpack.org/iris.csv",
    data, mlpack::NoFatal);
if (!success)

>     std::cout << "Error loading dataset" << std::endl;

Reviewer: ```suggestion
  std::cout << "Error loading dataset" << std::endl;
```
(style fix)

```
---

## Loading from remote URLs

mlpack supports loading datasets from URLs.  Files will be downloaded using the
[cpp-httplib](https://github.com/yhirose/cpp-httplib) library, which is bundled
with mlpack for ease of use.

When a remote URL is given to `Load()`:

 * The URL must start with either `http://` or `https://` or loading will fail.

 * If the URL starts with `https://`, then [`#define MLPACK_USE_HTTPS`](compile.md#configuring-mlpack-with-compile-time-definitions) should be
     declared before `#include <mlpack.hpp>`.

>  * For loading HTTPS data, support must be enabled with

Reviewer: ```suggestion
 * If the URL starts with `https://`, support must be enabled with
```

(or, "For loading data from an HTTPS source"; "HTTPS data" says that the type of the data is HTTPS, but that is not true, its source is HTTPS but its type is whatever the actual data is)

      << std::endl;
}
```

## Online data load example

> 

Author: nope I will remove it
